valora-inc/wallet
# Dynamic CICO Text And Post Exchange Send Flow

### Description

A PR to fix  _Withdraw flow is broken in Add/Withdraw menu_ #224.
- Includes send button from cash out exchange screens.
- Dynamic text based on selected currency in top text on cash out exchange screens.
- Remove Account Address from cash out exchange screens.
- Add Analytics for new flow

### Other changes

- Skipped Bidali cash out test as the custom webview was causing issues with detox.

### Tested

- [x] Tested locally on iOS
- [x] Tested locally on Android

### How others should test

#### Test case - cUSD Cryptocurrency Exchange

1. Go to Settings
2. Tap 'Add & Withdraw'
3. Tap 'Withdraw Funds'
4. Tap 'Celo Dollar (cUSD)' & 'Cryptocurrency exchange'
5. Tap 'Next'
6. Observe top text should say 'You can transfer cUSD to the following exchanges. Once you have retrieved your address for your chosen exchange, you can continue to the next step.
7. Tap 'Send'.
8. Check that you have arrived in the Send Flow and that the confirm your number prompt is not visible.
9. Enter an address - e.g. `0x22c8a9178841ba95a944afd1a1faae517d3f5daa` for Alfajores.
10. Accept send to unknown address prompt if served.
11. `Send cUSD`, cUSD Balance should be visible, and that user is unable to change to cEUR.
12. Complete send flow.

#### Test case - cEUR Cryptocurrency Exchange

1. Go to Settings
2. Tap 'Add & Withdraw'
3. Tap 'Withdraw Funds'
4. Tap 'Celo Euro (cEUR)' & 'Cryptocurrency exchange'
5. If in US, should see 'There are no exchanges available for cEUR in your country.', 'Switch Currency' option & 'Contact Support' option.
6. Check that options work accordingly.

#### Test case - CELO Cryptocurrency Exchange
1. Go to Settings
2. Tap 'Add & Withdraw'
3. Tap 'Withdraw Funds'
4. Tap 'Celo Euro (cEUR)' & 'Cryptocurrency exchange'
5. Should see a list of exchanges and a 'Withdraw CELO' button
6. Tap Withdraw CELO - Should enter send to address flow

### Screenshots
<div>
<img src="https://user-images.githubusercontent.com/26950305/135939932-9b2a9eea-9eea-48ff-9cbc-dc992e1d224c.png" width="19%" height="auto" />
<img src="https://user-images.githubusercontent.com/26950305/135939935-8fa960bb-36c4-4575-b3f5-c6f20a6d3db8.png" width="19%" height="auto" />
<img src="https://user-images.githubusercontent.com/26950305/136105758-afdb2c9c-0713-4940-bba5-0ad186e4b5df.png" width="19%" height="auto" />
<img src="https://user-images.githubusercontent.com/26950305/135939939-1e0bc823-4fb9-48f5-aecd-d80182653aa4.png" width="19%" height="auto" />
<img src="https://user-images.githubusercontent.com/26950305/136105717-8f454351-3aca-47ef-b58c-257db7c17075.png" width="19%" height="auto" />
</div>

### Related issues

- #224
- #921

### Backwards compatibility

This change is backwards compatible
